compiler/codegen/OMRCodeGenPhase.cpp

```diff
@@ -439,7 +439,9 @@ OMR::CodeGenPhase::performSetupForInstructionSelectionPhase(TR::CodeGenerator *
    {
    TR::Compilation *comp = cg->comp();
 
-   if (TR::Compiler->target.cpu.isZ() && TR::Compiler->om.shouldGenerateReadBarriersForFieldLoads())
+   if (TR::Compiler->target.cpu.isZ()
+           && !TR::Compiler->om.shouldReplaceGuardedLoadWithSoftwareReadBarrier()
+           && TR::Compiler->om.shouldGenerateReadBarriersForFieldLoads())
```

Comment:
Why do we still need to do this anchoring business at all for hardware/software? I thought we are deprecating this code for both hardware and software and using the new common lowering phase for generating read barriers?

Reply:
This has been removed.